nixos/modules/services/monitoring/datadog-agent.nix

```diff
@@ -201,6 +201,16 @@ in {
           excluded_interfaces = [ "lo" "lo0" ]; } ];
       };
     };
+
+    permissions = mkOption {
```

Comment:
If User/Group don't equal `datadog`, the `datadog` user shouldn't be created probably.

Reply:
Is there not a use case for setting the User/Group to root?

Reply:
Yes, but in either way a user named `datadog` will be created in the module and you probably don't want to create a new user that is unused then.

Reply:
yeah looks like you're right.
![datadog_user](https://user-images.githubusercontent.com/12566063/86064658-89b1b280-ba22-11ea-9393-7867622221cc.png)
I have this unused datadog user.

Maybe I'm asking the wrong question here. Perhaps it should be, **Why can't datadog access disk metrics?**

Reply:
Hey @Ma27 I did some digging. Here is what I have:

The datadog-agent's User/Group is `datadog` by default from the datadog module. The agent attempts to call into `run/user/1000/gvfs` or the Gnome Virtual File System. Take a look.
```
[myorg@myhost:/run/user]$ ll
total 0
drwx------ 5 myorg myorg 100 Jun 30 07:41 1000

[myorg@myhost:/run/user/1000]$ ll
total 0
drwx------ 3 myorg myorg 60 Jun 30 07:41 dbus-1
dr-x------ 2 myorg myorg  0 Jun 30 07:41 gvfs
drwxr-xr-x 2 myorg myorg 80 Jun 30 19:41 systemd
```

Of course it makes sense that the agent doesn't have access to `gvfs`. I also found similar issues [here](https://github.com/DataDog/datadog-agent/issues/5082) and [here](https://github.com/DataDog/dd-agent/issues/2932) where the fix is to either set root permissions or blacklist the offending error. So naturally I tried

```
[myorg@myhost:~]$ cat /etc/datadog-agent/conf.d/disk.d/conf.yaml | jq
{
  "init_config": null,
  "instances": [
    {
      "device_blacklist": [
        "gvfs"
      ],
      "file_system_blacklist": [
        "gvfs"
      ],
      "mount_point_blacklist": [
        "gvfs"
      ],
      "use_mount": false
    }
  ]
}

```

And, yet even after attempting to blacklist the gvfs I'm still left with the same error as before:
```
Jun 30 20:41:37 myhost somehash-unit-script-datadog-agent-start[32208]: 2020-06-30 20:41:37 UTC | CORE | WARN | (pkg/collector/py/datadog_agent.go:148 in LogMessage) | (disk.py:107) | Unable to get disk metrics for /run/user/1000/gvfs: [Errno 13] Permission denied: '/run/user/1000/gvfs'
```

I would agree that datadog shouldn't be run as root but the only solution that has worked for me thus far is modifying upstream to run datadog via root.

Reply:
* Without taking a look at the details, you may want to take a look at the parameters in [`systemd.exec(5)`](https://www.freedesktop.org/software/systemd/man/systemd.exec.html)
* I'm afraid that you're misunderstanding my point: the module *creates* a new user named *datadog*. But this shouldn't be done IMHO *if* you declare a user which is not named `datadog`.

Reply:
I understand your point. The module creates a datadog user. Therefore it would be unwise to use the module and then set the User/Group to some other user i.e. `root`. 

Then it comes down to getting the datadog user to access the `/run/user/1000/gvfs` which is owned by my user named `myorg`.

Reply:
how about adding the `datadog` group as member to the `gvfs` group?